File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/retrieval/GetAllStoredFilesInLibrary.java
package com.lasthopesoftware.bluewater.client.stored.library.items.files.retrieval;

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId;
import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile;
import com.namehillsoftware.handoff.promises.Promise;

import java.util.Collection;

public interface GetAllStoredFilesInLibrary {
	Promise<Collection<StoredFile>> promiseAllStoredFiles(LibraryId libraryId);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/retrieval/GetAllStoredFilesInLibrary.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files.retrieval

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile
import com.namehillsoftware.handoff.promises.Promise

interface GetAllStoredFilesInLibrary {
    fun promiseAllStoredFiles(libraryId: LibraryId): Promise<Collection<StoredFile>>
}
